src/natcap/invest/urban_flood_risk_mitigation.py

```diff
@@ -322,107 +313,245 @@ def execute(args):
+        # Reproject the built infrastructure vector to the target SRS.
+        reprojected_structures_path = os.path.join(
+                intermediate_dir, 'structures_reprojected.gpkg')
+        reproject_built_infrastructure_task = task_graph.add_task(
+            func=pygeoprocessing.reproject_vector,
+            args=(args['built_infrastructure_vector_path'],
+                  target_sr_wkt,
+                  reprojected_structures_path),
+            kwargs={'driver_name': 'GPKG'},
+            target_path_list=[reprojected_structures_path],
+            task_name='reproject built infrastructure to target SRS')
+
+        # determine the total damage to all infrastructure in the watershed/AOI
+        damage_to_infrastructure_in_aoi_task = task_graph.add_task(
+            func=_calculate_damage_to_infrastructure_in_aoi,
+            args=(reprojected_aoi_path,
+                  reprojected_structures_path,
+                  args['infrastructure_damage_loss_table_path']),
+            dependent_task_list=[
+                reprojected_aoi_task,
+                reproject_built_infrastructure_task],
+            task_name='calculate damage to infrastructure in aoi')
+        summary_tasks.append(damage_to_infrastructure_in_aoi_task)
+
+        # Determine flood_volume over the watershed
+        flood_volume_in_aoi_task = task_graph.add_task(
+            func=pygeoprocessing.zonal_statistics,
+            args=(
+                (runoff_retention_raster_path, 1),
+                reprojected_aoi_path),
+            dependent_task_list=[flood_vol_task],
+            task_name='zonal_statistics over the flood_volume raster')
+        flood_volume_stats = flood_volume_in_aoi_task.get()
+        damage_per_aoi_stats = damage_to_infrastructure_in_aoi_task.get()
+
+    summary_vector_path = os.path.join(
+        args['workspace_dir'], 'flood_risk_service%s.shp' % file_suffix)
+    write_summary_vector_stats = task_graph.add_task(
+        func=_write_summary_vector,
+        args=(reprojected_aoi_path,
+              summary_vector_path),
+        kwargs={
+            'runoff_ret_stats': runoff_retention_stats_task.get(),
+            'runoff_ret_vol_stats': runoff_retention_volume_stats_task.get(),
+            'damage_per_aoi_stats': damage_per_aoi_stats,
+            'flood_volume_stats': flood_volume_stats,
+        },
+        target_path_list=[summary_vector_path],
+        task_name='write summary stats to flood_risk_service.shp',
+        dependent_task_list=summary_tasks)
 
     task_graph.close()
     task_graph.join()
 
 
-def _pickle_zonal_stats(
-        base_vector_path, base_raster_path, target_pickle_path):
-    """Calculate Zonal Stats for a vector/raster pair and pickle result.
+def _write_summary_vector(
+        source_aoi_vector_path, target_vector_path, runoff_ret_stats=None,
+        runoff_ret_vol_stats=None, damage_per_aoi_stats=None,
+        flood_volume_stats=None):

```

Comment:
How about a docstring for this function?

Reply:
Can't believe I missed that.  Added in ebde3b71.